src/coreclr/jit/lsra.cpp

```diff
@@ -714,6 +714,9 @@ LinearScan::LinearScan(Compiler* theCompiler)
         availableIntRegs &= ~RBM_CALLEE_SAVED | RBM_ENC_CALLEE_SAVED;
         availableFloatRegs &= ~RBM_CALLEE_SAVED;
         availableDoubleRegs &= ~RBM_CALLEE_SAVED;
+#if defined(TARGET_XARCH)
+        availableMaskRegs &= ~RBM_CALLEE_SAVED;
```

Comment:
Today, no. But it doesn't hurt anything to have it and likewise aren't filtering other scenarios where we may not have any callee saved for the register type.

If we were to do something here, we should probably change it in a follow up PR to only mask off the relevant registers.

That is, `availableFloatRegs` should use `~RBM_FLT_CALLEE_SAVED` since the `RBM_CALLEE_SAVED` bits set by  `RBM_INT_CALLEE_SAVED` will never be hit.

That in turn would allow the compiler to optimize these away for platforms that have no callee saved registers for a given kind (like unix for float).